vertical-pod-autoscaler/pkg/utils/metrics/updater/updater.go

```diff
@@ -41,16 +85,77 @@ var (
 
 // Register initializes all metrics for VPA Updater
 func Register() {
-	prometheus.MustRegister(evictedCount)
-	prometheus.MustRegister(functionLatency)
+	prometheus.MustRegister(controlledCount, evictableCount, evictedCount, vpasWithEvictablePodsCount, vpasWithEvictedPodsCount, functionLatency)
 }
 
 // NewExecutionTimer provides a timer for Updater's RunOnce execution
 func NewExecutionTimer() *metrics.ExecutionTimer {
 	return metrics.NewExecutionTimer(functionLatency)
 }
 
-// AddEvictedPod increases the counter of pods evicted by VPA
-func AddEvictedPod() {
-	evictedCount.Add(1)
+// newSizeBasedGauge provides a wrapper for counting items in a loop
+func newSizeBasedGauge(gauge *prometheus.GaugeVec) *SizeBasedGauge {
+	obj := SizeBasedGauge{
+		values: make(map[int]int),
```

Comment:
could just be an int array, initialized to 0 values

Reply:
Done.